src/App/Pages/Accounts/SetPasswordPageViewModel.cs

```diff
@@ -256,6 +247,23 @@ public async Task SubmitAsync()
             }
         }
 
+        private async Task<KeysRequest> GetKeysForSetPasswordRequestAsync(UserKey newUserKey)
+        {
+            KeysRequest keysRequest = null;
+            if (ForceSetPasswordReason != ForcePasswordResetReason.TdeUserWithoutPasswordHasPasswordResetPermission)
+            {
+                var (newPublicKey, newProtectedPrivateKey) = await _cryptoService.MakeKeyPairAsync(newUserKey);
+                keysRequest = new KeysRequest
+                {
+                    PublicKey = newPublicKey,
+                    EncryptedPrivateKey = newProtectedPrivateKey.EncryptedString
+                };
+
+            }
+
+            return keysRequest;
+        }
```

Comment:
🎨 The code can be improved like this to avoid defining `keysRequest`:
```suggestion
        private async Task<KeysRequest> GetKeysForSetPasswordRequestAsync(UserKey newUserKey)
        {
            if (ForceSetPasswordReason == ForcePasswordResetReason.TdeUserWithoutPasswordHasPasswordResetPermission)
            {
                return null;
            }

            var (newPublicKey, newProtectedPrivateKey) = await _cryptoService.MakeKeyPairAsync(newUserKey);
            return new KeysRequest
            {
                PublicKey = newPublicKey,
                EncryptedPrivateKey = newProtectedPrivateKey.EncryptedString
            };
        }
```